hudi-client/src/main/java/org/apache/hudi/io/HoodieCommitArchiveLog.java

```diff
@@ -268,6 +270,19 @@ public Path getArchiveFilePath() {
     return archiveFilePath;
   }
 
+  private void writeHeaderBlock(Schema wrapperSchema, List<HoodieInstant> instants) throws Exception {
+    if (!instants.isEmpty()) {
+      Collections.sort(instants, HoodieInstant.COMPARATOR);
+      HoodieInstant minInstant = instants.get(0);
+      HoodieInstant maxInstant = instants.get(instants.size() - 1);
+      Map<HeaderMetadataType, String> metadataMap = Maps.newHashMap();
+      metadataMap.put(HeaderMetadataType.SCHEMA, wrapperSchema.toString());
+      metadataMap.put(HeaderMetadataType.MIN_INSTANT_TIME, minInstant.getTimestamp());
+      metadataMap.put(HeaderMetadataType.MAX_INSTANT_TIME, maxInstant.getTimestamp());
+      this.writer.appendBlock(new HoodieAvroDataBlock(Collections.emptyList(), metadataMap));
+    }
+  }
+
   private void writeToFile(Schema wrapperSchema, List<IndexedRecord> records) throws Exception {
```

Comment:
Move the writing of the header to this part, basically, augment the same DataBlock that is has the archived records with the metadata information that you want to push here, we already write the schema, just add more entries (like above) to the headers here. Then you will be able to read each block and then filter based on whether the block should be considered or not - this is more generic than adding an extra empty log block to track min/max over the entire file (which is hard since the file keeps growing anyways)

Reply:
I've included decision for including header block above. Let me know. file is closed after archiving all instants that qualify. So i think file can grow is not a issue. Correct me if i'm reading this wrong.

Reply:
You are right that the file is closed after archiving all instants that qualify in that archiving process. But the next time an archival kicks in, it will check if the archival file is grown to a certain size (say 1GB), if not, it will append the next archival blocks to the same file..

Reply:
addressed. added a unit test for this scenario